plugins/woocommerce/tests/php/includes/class-wc-order-item-product-test.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Unit tests for the WC_Order_Item_Product class functionalities.
+ *
+ * @package WooCommerce\Tests
+ */
+
+declare( strict_types=1 );
+
+use WC_Helper_Product;
+use WC_Helper_Order;
+use WC_Order_Item_Product;
+use WC_Unit_Test_Case;
+
+/**
+ * WC_Order_Item_Product unit tests.
+ */
+class WC_Order_Item_Product_Test extends WC_Unit_Test_Case {
+
+	/**
+	 * Test that backorder meta is excluded from formatted meta data
+	 * for completed orders.
+	 */
+	public function test_get_formatted_meta_data_excludes_backorder_on_completed() {
+		// 1. Setup: Create product and set backorders allowed.
+		$product = WC_Helper_Product::create_simple_product();
+		$product->set_manage_stock( true );
+		$product->set_stock_quantity( 0 );
+		$product->set_backorders( 'notify' ); // Allow backorders with notification.
+		$product->save();
+
+		// 2. Create Order and add product item.
+		$order = WC_Helper_Order::create_order();
+		$item  = new WC_Order_Item_Product();
+		$item->set_product( $product );
+		$item->set_quantity( 1 );
+		$item->set_order_id( $order->get_id() );
+
+		// 3. Add backorder meta.
+		$backorder_meta_key = 'Backordered';
+		$item->add_meta_data( $backorder_meta_key, 1, true ); // Value typically is the backordered quantity.
+		$item->save();
+		$order->add_item( $item );
+		$order->save();
+
+		// 4. Assert: Check meta exists before completion.
+		$item_id                = $item->get_id();
+		$order_item             = $order->get_item( $item_id ); // Re-fetch item from order.
+		$formatted_meta_before  = $order_item->get_formatted_meta_data();
+		$found_backorder_before = false;
+		foreach ( $formatted_meta_before as $meta ) {
+			if ( $meta->key === $backorder_meta_key ) {
+				$found_backorder_before = true;
+				break;
+			}
+		}
+		$this->assertTrue( $found_backorder_before, 'Backorder meta should exist before order completion.' );
+
+		// 5. Complete the order.
+		$order->update_status( 'completed' );
```

Comment:
We can also use here the enum class `\Automattic\WooCommerce\Enums\OrderStatus`. What do you think?